examples/sampling/pyg/node_classification.py

```diff
@@ -0,0 +1,234 @@
+        optimizer.zero_grad()
+        out = model(minibatch.blocks, node_features, device)
+        loss = criterion(out, labels)
+        total_loss += loss.item()
+        total_correct += MF.accuracy(
+            out, labels, task="multiclass", num_classes=num_classes
+        ) * labels.size(0)
+        total_samples += labels.size(0)
+        loss.backward()
+        optimizer.step()
+        num_batches += 1
+    avg_loss = total_loss / num_batches
+    avg_accuracy = total_correct / total_samples
+    return avg_loss, avg_accuracy
+
+
+@torch.no_grad()
+def evaluate(model, dataloader, device, num_classes):
+    model.eval()
+    y_hats = []
+    ys = []
+    for minibatch in dataloader:
+        node_features = minibatch.node_features["feat"]
+        labels = minibatch.labels
+        out = model(minibatch.blocks, node_features, device)
+        y_hats.append(out)
+        ys.append(labels)
+
+    return MF.accuracy(
+        torch.cat(y_hats),
+        torch.cat(ys),
+        task="multiclass",
+        num_classes=num_classes,
+    )
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Which dataset are you going to use?"
+    )
+    parser.add_argument(
+        "--dataset",
+        type=str,
+        default="ogbn-arxiv",
+        help='Name of the dataset to use (e.g., "ogbn-products", "ogbn-arxiv")',
+    )
+    args = parser.parse_args()
+    dataset_name = args.dataset
+    dataset = gb.BuiltinDataset(dataset_name).load()
+    graph = dataset.graph
+    feature = dataset.feature
+    train_set = dataset.tasks[0].train_set
+    valid_set = dataset.tasks[0].validation_set
+    test_set = dataset.tasks[0].test_set
+    num_classes = dataset.tasks[0].metadata["num_classes"]
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+    train_dataloader = create_dataloader(
+        train_set, graph, feature, device, is_train=True
+    )
+    valid_dataloader = create_dataloader(
+        valid_set, graph, feature, device, is_train=False
+    )
+    test_dataloader = create_dataloader(
+        test_set, graph, feature, device, is_train=False
+    )
+    in_channels = feature.size("node", None, "feat")[0]
+    hidden_channels = 128
+    model = GraphSAGE(in_channels, hidden_channels, num_classes).to(device)
+    optimizer = torch.optim.Adam(model.parameters(), lr=0.01, weight_decay=5e-4)
+    criterion = torch.nn.CrossEntropyLoss()
+    for epoch in range(10):
+        train_loss, train_accuracy = train(
+            model, train_dataloader, optimizer, criterion, device, num_classes
+        )
+
+        valid_accuracy = evaluate(model, valid_dataloader, device, num_classes)
+        test_accuracy = evaluate(model, test_dataloader, device, num_classes)
+        print(
+            f"Epoch {epoch}, Train Loss: {train_loss:.4f}, Train Accuracy: {valid_accuracy:.4f}, "
+            f"Valid Accuracy: {valid_accuracy:.4f}, Test Accuracy: {test_accuracy:.4f}"
+        )
```

Comment:
```suggestion
        print(
            f"Epoch {epoch}, Train Loss: {train_loss:.4f}, Train Accuracy: {valid_accuracy:.4f}, "
            f"Valid Accuracy: {valid_accuracy:.4f}"
        )
    test_accuracy = evaluate(model, test_dataloader, device, num_classes)
    print(f"Test Accuracy: {test_accuracy:.4f}")
```
Let's compute the test accuracy only at the end of training.

Reply:
You can commit this fix with your own username/e-mail and git will show your name for the whole PR.

Reply:
I ran the code and noticed that the reason why the products dataset takes so long is because the test accuracy is computed at every epoch.

Reply:
right, thank you so much!